.github/workflows/build-test.yml

```diff
@@ -56,10 +58,10 @@
       # Portals won't build without a configuration. For now, copy the test configuration.
       # We may want to invert the model and give each portal its own project in the workspace, so we can build the portals in parallel.
      - run: pnpm nx run portals:copy-test-configuration
-     - run: pnpm build
-     - run: pnpm lint
+     - run: pnpm nx affected --target=build --parallel=2 --base=remotes/origin/main
+     - run: pnpm nx affected --target=lint --parallel=2 --base=remotes/origin/main
      - id: test
-       run: pnpm test
+       run: pnpm nx affected --target test --parallel=2 --base=remotes/origin/main
```

Comment:
With `affected`, if you just change one project, the target job will only run for that project and its dependents.